core/mo-services-impl/ccsds-com-impl/src/main/java/esa/mo/com/impl/archive/db/CallableGetCOMObjects.java

```diff
@@ -43,23 +43,27 @@ public List<COMObjectEntity> call() {
 
         try {
             ResultSet rs;
+            String query = this.transactionsProcessor.dbBackend.getPreparedStatements()
+                                                               .getSelectCOMObjectsQueryString();
             if (this.transactionsProcessor.dbBackend.isPostgres) {
                 // Array-bind is supported in Postgres
-                PreparedStatement stmt = this.transactionsProcessor.dbBackend.getPreparedStatements()
-                        .getSelectCOMObjects();
+                PreparedStatement stmt = c.prepareStatement(query);
                 Array idsArray = c.createArrayOf("BIGINT", ids.toArray());
                 stmt.setInt(1, objTypeId);
                 stmt.setInt(2, domainId);
                 stmt.setArray(3, idsArray);
                 rs = stmt.executeQuery();
             } else {
-                String idsString = ids.stream().map(Object::toString).collect(Collectors.joining(", "));
-                String query = "SELECT objectTypeId, domainId, objId, "
-                        + "timestampArchiveDetails, providerURI, network, sourceLinkObjectTypeId, "
-                        + "sourceLinkDomainId, sourceLinkObjId, relatedLink, objBody " + "FROM COMObjectEntity "
-                        + "WHERE ((objectTypeId = " + objTypeId + ") AND (domainId = " + domainId + ") AND (objId in ("
-                        + idsString + ")))";
-                rs = c.createStatement().executeQuery(query);
+                String parameters = ids.stream().map(id -> "?").collect(Collectors.joining(", "));
+                query = String.format(query, parameters);
+                PreparedStatement stmt = c.prepareStatement(query);
```

Comment:
See the above comment. In general code looks good, but do not use prepared statements for a statement which changes every query -> rather call `createStatement().executeQuery()` API instead